ITE/10/README.adoc

```diff
@@ -0,0 +1,287 @@
+checkout step.
+
+=== Setting Predicate Type Expectations for Steps and Inspections
+
+Currently, step declarations have the following schema.
+
+```yaml
+name: string
+threshold: int
+expectedMaterials: list of artifact rules
+expectedProducts: list of artifact rules
+pubkeys: list of authorized key IDs
+expectedCommand: list of strings
+```
+
+This ITE proposes updating this schema to incorporate predicate types.
+
+```yaml
+name: string
+command: list of strings
+expectedMaterials: list of artifact rules
+expectedProducts: list of artifact rules
+expectedPredicates: list of expectedStepPredicates
+```
+
+Each `expectedStepPredicate` object has the following schema.
+
+```yaml
+predicateType: string
+functionaries: list of authorized functionary key IDs
+threshold: int
+```
+
+The `predicateType` field contains the matching `TypeURI` of the predicate
+expected for the step. The original `pubkeys` field is updated to
+`functionaries` to generalize signing semantics. Finally, `threshold` applies
+per predicate type rather than the step as a whole. Note that this schema
+supports the in-toto Attestation Framework while retaining all the capabilities
+of in-toto 1.0 layouts.
+
+This change maintains the core model of the original in-toto layout while
+allowing for multiple predicate types to be associated for a single step. Note
+that while this schema allows for multiple predicate types, the expectation is
+that they share the same materials and products--if two attestations do not
+share the input and output artifacts, they must belong to different steps. Of
+course, as informational predicates do not result in products, the
+`expectedProducts` field does not apply to them.  In a step with exclusively
+informational predicates, the `expectedProducts` field may be omitted or left
+empty. In a step with a mixture of transformational and informational
+predicates, the field only applies to the transformational predicates. The
+`expectedMaterials` field functions as it currently does, and applies to all
+predicates, informational and transformational.
+
+=== Supporting Sublayouts
+
+The in-toto specification has support for sublayouts which are steps in a
+layout that have not links but a bundle of another layout and links which are
+recursively verified first. In in-toto v1.0, a sublayout is declared to the
+verification workflow by presenting the sublayout file in place of the link
+expected for the corresponding step.
+
+The updated in-toto layout specified in this ITE is similar. The step is
+declared with an expected predicate type and when a layout is encountered, the
+verifier recursively verifies the sublayout, generating the expected predicate
+in the process. Note that this expects the verifier to be capable of generating
+distinct predicates--the in-toto v1.0 verifier was only aware of links and
+sublayout verification therefore generated an in-toto summary link. This is
+discussed further on.
```

Comment:
If possible, it would be helpful to forward-reference to the "Custom Verifiers" section here.